### ISSUE TITLE
Incorrect python typehint for `add_items`

### ISSUE THREAD

**Issue by markkohdev** [spotify/voyager#23]
The python typehint for `add_items` (at least in IntelliJ) incorrectly names the second parameter `numpy_float32` instead of `ids`
![image](https://github.com/spotify/voyager/assets/1663379/423256da-a350-41f5-ad6a-a57459423cdc)

I'm guessing this is due to something wonky in either our bindings or in pybind11.  I also noticed that there's a tiny typo in the docstring for the method -- `id -> ids`
https://github.com/spotify/voyager/blob/main/python/bindings.cpp#L433C7-L433C7

**Comment by markkohdev** [comment]
Looks like this happens with `query()` as well
![image](https://github.com/spotify/voyager/assets/1663379/f0e7c7fd-f4ad-4e7b-9f15-8c86c1c81707)

**Comment by eXamadeus** [comment]
I don't know if this is related, but when you use the static version, it also complains since it thinks that the first argument should be `self` even though the `add_items` and certain other functions are, in fact, statically available as class methods.

Ex:

```
from voyager import Index

loaded = Index.load(f"indexes/{path}.voy")
```

<img width="523" alt="image" src="https://github.com/spotify/voyager/assets/37161095/75c1a543-4407-4884-b874-ea7771fdbbf7">

It seems to only have type hints for instances. Then again, I haven't written production code for python in years (before type hints were ubiquitous) so I might just have something configured poorly.

<img width="629" alt="image" src="https://github.com/spotify/voyager/assets/37161095/87bb931b-92d9-4004-96d2-53e20549c0b4">

**Comment by markkohdev** [comment]
@eXamadeus thanks for highlighting this!  Your issue isn't exactly the same thing, but it's close enough that I can try to fix both of these at once when I'm done with my current improvement.

For context, since we use [python _bindings_ written in c++](https://github.com/spotify/voyager/blob/main/python/bindings.cpp) to call out to the core C++ code, we actually have to _generate_ these typehints based on the c++ binding code.  We do that with [this script](https://github.com/spotify/voyager/blob/main/python/scripts/generate_type_stubs_and_docs.py) and so somewhere in the process the staticmethod indicator likely isn't getting acknowledged.  The binding code itself does contain [an indicator that the method is static](https://github.com/spotify/voyager/blob/main/python/bindings.cpp#L916C1-L916C20) so we should be able to populate that accordingly.

**Comment by eXamadeus** [comment]
Yeah what I noticed definitely isn't _directly_ related, at best it is ancillary since it relates to the bindings. Either way, thank you for the context.

I'll poke around in my freetime to see if I can help. I can't promise much, since it's been even longer since I've written any C++, haha.
